### PR TITLE
Fix memory limit check bytes to GB conversion

### PR DESCRIPTION
**Pull request by 1pkg**
## Motivation/summary

This PR closed the issue https://github.com/elastic/apm-server/issues/13837, by fixing the bytes to GB conversion in the fallback memory check.

## Checklist

~~Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)~~
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

The easiest way to test this change is to create a custom cgroup with a memory limit that is larger than the total system memory, then start apm-server binary with the created cgroup. Observer APM Server logs for "cgroup memory limit exceed available memory, falling back to the total system memory".

```bash
cgcreate  -g memory:apm_test
cgget -r memory.limit_in_bytes=10737418240 /apm_test
cgexec -g memory:apm_test ./apm-server run
```

## Changelog

==== Bug fixes

- Fix memory limit fallback check bytes to GB conversion {pull}13838[13838]

### CODE REVIEW

File: internal/beater/beater.go
Comment (1pkg) [reply internal/beater/beater.go]: Thank you for a prompt review, I agree simplifying and not repeating conversion multiple times is a good idea. Pushed the updates in the last commit.

**Review by marclop — DISMISSED** [review]
Thanks for the fix and tidying the code!

File: internal/beater/beater.go
Comment (carsonip) [inline internal/beater/beater.go:213]: nit: the magic number is slightly hard to read
```suggestion
	var memLimitGB = float64(memLimit) / (1<<30)
```

File: changelogs/8.15.asciidoc
Comment (carsonip) [inline changelogs/8.15.asciidoc:39]: This section is for 8.15.0. Section for 8.15.1 is not ready yet. If this fix is planned 8.15.1, let's keep this PR open until 8.15.0 release and we have a 8.15.1 section in the changelog for easier backport.

**Review by carsonip — COMMENTED** [review]
non-blocker: is this something that can be covered by tests?

File: changelogs/8.15.asciidoc
Comment (carsonip) [reply changelogs/8.15.asciidoc:39]: I always forget about this: changelogs in release notes for patch releases are usually added right before release, instead of added at PR merge time unlike for usual PRs. If you could move the changelog entry to PR description, the release manager would pick it up at 8.15.1 release.

Comment (1pkg) [reply changelogs/8.15.asciidoc:39]: Sure, thank you for raising this, will update it.

File: internal/beater/beater.go
Comment (1pkg) [reply internal/beater/beater.go]: updated

File: changelogs/8.15.asciidoc
Comment (1pkg) [reply changelogs/8.15.asciidoc]: updated the changelogs file

**Review by carsonip — COMMENTED** [review]
changes lgtm. Thanks for the tests.

Could you remove the changelog? It will be added back later in the release process. Sorry if I didn't express that clearly in the earlier comment.

**Review by carsonip — APPROVED** [review]
I'll approve it first so you can merge it right away after removing the changelog (if github allows you to do that)

**Review by carsonip — APPROVED** [review]

**Pull request merged**
